src/Serializer/AbstractItemNormalizer.php

```diff
@@ -285,7 +285,9 @@ private function denormalizeRelation(string $attributeName, PropertyMetadata $pr
             }
         }
 
-        if (!$this->resourceClassResolver->isResourceClass($className) || $propertyMetadata->isWritableLink()) {
+        $isResourceClass = $this->resourceClassResolver->isResourceClass($className);
```

Comment:
If I understand correctly, the variables are useless, can you just add the `is_array` check without introducing new vars?